src/components/List/List.tsx

```diff
@@ -1,90 +1,157 @@
+  // searchTerm?: string;
+  // /** Defines if this is searchable list or not **/
+  // isSearchable?: boolean;
+} & Omit<AriaListBoxProps<SelectOption>, 'selectionMode' | 'onSelectionChange' | 'children'> &
+  TestProps &
+  Omit<React.InputHTMLAttributes<HTMLUListElement>, 'onChange'>;
 
-const List = React.forwardRef<HTMLUListElement, ListProps>(
-  (
-    {
-      data,
-      rowSize,
-      width,
-      height,
-      isVirtualized = false,
-      selectedItem,
-      isSearchable,
-      defaultOption,
-      searchTerm,
-      handleOptionClick,
-      dataTestId,
-      ...rest
-    },
-    ref
-  ) => {
-    return (
-      <div css={wrapperStyle({ width, isSearchable })}>
-        {isVirtualized ? (
-          <VirtualizedList
-            items={data}
-            rowSize={rowSize}
-            customWidth={width}
-            customHeight={height}
-            ref={ref}
-            selectedItem={selectedItem}
-            defaultOption={defaultOption}
-            searchTerm={searchTerm}
-            handleOptionClick={handleOptionClick}
-            dataTestId={dataTestId}
-            {...rest}
-          />
-        ) : (
-          <NormalList
-            items={data}
-            rowSize={rowSize}
-            width={width}
-            height={height}
-            ref={ref}
-            selectedItem={selectedItem}
-            isSearchable={isSearchable}
-            defaultOption={defaultOption}
-            searchTerm={searchTerm}
-            handleOptionClick={handleOptionClick}
-            dataTestId={dataTestId}
-            {...rest}
-          />
-        )}
+const List = React.forwardRef<HTMLUListElement, ListProps>((props, ref) => {
+  const {
+    width,
+    height,
+    isVirtualized = false,
+    // isSearchable,
+    // searchTerm,
+    dataTestId,
+  } = props;
+  /** ignore the change on the `onSelectionChange` from any to unknown **/
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const state = useListState<SelectOption>({ ...props, selectionMode: 'single' });
+  const myRef = React.useRef(null);
+  const combinedRefs = useCombinedRefs(myRef, ref);
+  /** ignore the change on the `onSelectionChange` from any to unknown **/
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const { listBoxProps } = useListBox(props, state, combinedRefs);
+
+  const firstKey = state.collection.getFirstKey();
+  const first = firstKey ? state.collection.getItem(firstKey) : null;
+
+  return (
+    <div css={wrapperStyle({ width })}>
+      <div data-testid={dataTestId ? `${dataTestId}_list` : 'ictinus_list'}>
+        <div style={{ height: height || '100%' }}>
+          <Window
+            {...listBoxProps}
+            css={listStyle({ width, height })}
+            id={listBoxProps.id}
+            isVirtualizationEnabled={isVirtualized}
+            rowHeight={first?.props.rowSize === 'compact' ? 40 : 56}
```

Comment:
these values should be constants for better maintenance

Reply:
this is not deep comparison or anything, it would be one number that will be the same in every render. I don't see any performance implication here

Reply:
Im not saying this for performance reasons, but for maintenance. This numbers are defined by the designs so they should be kept stored in a config file; so if one day the height is changed in the design, the changes in the code will be easier if these constants are stored in one global config file.

Reply:
fixed [`0363daa` (#655)](https://github.com/Orfium/orfium-ictinus/pull/655/commits/a81f7514532e8bd05a3f1bc8474b515478ab1b39)